plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer.ts

```diff
@@ -570,32 +576,35 @@ export class SessionRecordingIngester {
     }
 
     public async stop(): Promise<PromiseSettledResult<any>[]> {
-        status.info('🔁', 'blob_ingester_consumer - stopping')
-        this.isStopping = true
+        if (!this.stoppingPromise) {
+            status.info('🔁', 'blob_ingester_consumer - stopping')
+            this.stoppingPromise = (async () => {
+                // NOTE: We have to get the partitions before we stop the consumer as it throws if disconnected
+                const assignedPartitions = this.assignedTopicPartitions
 
-        // NOTE: We have to get the partitions before we stop the consumer as it throws if disconnected
-        const assignedPartitions = this.assignedTopicPartitions
-        // Mark as stopping so that we don't actually process any more incoming messages, but still keep the process alive
-        await this.batchConsumer?.stop()
+                // There is a race between the revoke callback and this function - Either way one of them gets there and covers the revocations
+                void this.scheduleWork(this.realtimeManager.unsubscribe())
+                void this.scheduleWork(this.flushPartitions(assignedPartitions.map((x) => x.partition)))
```

Comment:
do we not need to add `commitAllOffsets` as well? my understanding is flushing doesn't necessarily commit, especially with the offset store

Reply:
Thats done in the flushPartitons command.